### PR TITLE
provider/aws: Update `aws_emr_cluster` to support Security Configuration

### PR DESCRIPTION
**Pull request by catsby**
Supersedes #13620 by adding a test, and also updating the documentation to specify the required EMR release label version

### CODE REVIEW

**Comment by mbfrahry** [comment]
Can we update a security configuration?

**Comment by catsby** [comment]
> Can we update a security configuration? 
@mbfrahry Security Configurations are themselves all `ForceNew`. Updating your Security Configuration will `ForceNew` here as well; EMR does not have any API for modifying that

**Comment by mbfrahry** [comment]
Can we update a security configuration?

File: website/source/docs/providers/aws/r/emr_cluster.html.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ The following arguments are supported:
 * `release_label` - (Required) The release label for the Amazon EMR release
 * `master_instance_type` - (Required) The EC2 instance type of the master node
 * `service_role` - (Required) IAM role that will be assumed by the Amazon EMR service to access AWS resources
+* `security_configuration` - (Optional) The security configuration name to attach to the EMR cluster, the security configuration cannot be created through Terraform. Only valid for EMR clusters with `release_label` 4.8.0 or greater
```
Comment (mbfrahry) [inline website/source/docs/providers/aws/r/emr_cluster.html.md]: Why can't the security_configuration be created through Terraform?  We create one for the test https://github.com/hashicorp/terraform/pull/14133/files#diff-9907e8ff4ea2f3a2f2b1fc56afbb49e5R1206

Comment (catsby) [inline website/source/docs/providers/aws/r/emr_cluster.html.md]: derp, forgot to remove that. That's from the old PR I took over, which didn't include the resource, and was created before I added the resource last Friday :D

**Comment by mbfrahry** [comment]
LGTM!

**Pull request merged**